tests/components/mcp_server/test_http.py

```diff
@@ -0,0 +1,345 @@
+
+    async_expose_entity(hass, CONVERSATION_DOMAIN, TEST_ENTITY, True)
+
+
+async def test_http_sse(
+    hass: HomeAssistant,
+    setup_integration: None,
+    hass_client: ClientSessionGenerator,
+) -> None:
+    """Test SSE endpoint can be used to receive MCP messages."""
+
+    client = await hass_client()
+
+    # Start an SSE session
+    response = await client.get(SSE_API)
+    assert response.status == HTTPStatus.OK
+    sse_iter = aiter(response.content)
+
+    # Decode a single SSE response that sends the messages endpoint
+    # event: endpoint
+    # data: /mcp_server/messages/session-id
+    line = (await anext(sse_iter)).decode()
+    assert line == EXPECTED_ENDPOINT_EVENT
+    line = (await anext(sse_iter)).decode()
+    assert line.startswith(EXPECTED_ENDPOINT_DATA)
+    endpoint_url = line[6:].strip()
+    line = (await anext(sse_iter)).decode()
+    assert line == "\r\n"
+
+    # Send an initialize message on the messages endpoint
+    response = await client.post(endpoint_url, json=INITIALIZE_MESSAGE)
+    assert response.status == HTTPStatus.OK
+
+    # Decode the initialize response event message from the SSE stream
+    line = (await anext(sse_iter)).decode()
+    assert line == EXPECTED_ENDPOINT_MESSAGE
+    line = (await anext(sse_iter)).decode()
+    assert line.startswith(DATA_PREFIX)
+    data = json.loads(line[6:])
+    assert data.get("jsonrpc") == "2.0"
+    assert data.get("id") == "request-id-1"
+    assert "serverInfo" in data.get("result", {})
+    assert "protocolVersion" in data.get("result", {})
+    line = (await anext(sse_iter)).decode()
+    assert line == "\r\n"
+
+
+async def test_http_messages_missing_session_id(
+    hass: HomeAssistant,
+    setup_integration: None,
+    hass_client: ClientSessionGenerator,
+) -> None:
+    """Test the tools list endpoint."""
+
+    client = await hass_client()
+    response = await client.post(MESSAGES_API.format(session_id="invalid-session-id"))
+    assert response.status == HTTPStatus.NOT_FOUND
+    response_data = await response.text()
+    assert response_data == "Could not find session ID 'invalid-session-id'"
+
+
+async def test_http_messages_invalid_message_format(
+    hass: HomeAssistant,
+    setup_integration: None,
+    hass_client: ClientSessionGenerator,
+) -> None:
+    """Test the tools list endpoint."""
+
+    client = await hass_client()
+    response = await client.get(SSE_API)
+    assert response.status == HTTPStatus.OK
+    it = aiter(response.content)
+    line = (await anext(it)).decode()
+    assert line == EXPECTED_ENDPOINT_EVENT
+    line = (await anext(it)).decode()
+    assert line.startswith(EXPECTED_ENDPOINT_DATA)
+    endpoint_url = line[6:].strip()
+
+    response = await client.post(endpoint_url, json={"invalid": "message"})
+    assert response.status == HTTPStatus.BAD_REQUEST
+    response_data = await response.text()
+    assert response_data == "Could not parse message"
+
+
+async def test_http_sse_multiple_config_entries(
```

Comment:
technically not possible because of the key in `manifest.json` that only a single entry is allowed.

Reply:
(but ok for coverage 👌 )

Reply:
yes, this is added for coverage. I added a comment to the function stating so.